### ISSUE TITLE
Firestore - .orderBy() with collectionReference throws an error

### ISSUE THREAD

**Issue by OhadRon** [invertase/react-native-firebase#1132]
### Issue

Whenever I call `.onSnapshot(...` on a `collectionReference` after calling an `.orderBy(...` I get an error: 
```
com.facebook.react.bridge.ReadableNativeMap cannot be cast to java.util.Map

applyOrders
RNFirebaseFirestoreCollectionReference.java:190
...
```
This happens only on Android. Removing the `orderBy(...` clause makes everything work great again.

This started happening once I upgraded from 4.0.4 - happens on 4.1.0 and on 4.2.0, so I tried to follow the installation instructions in the 4.1.0 release notes:
1. Updated to `com.android.tools.build:gradle:3.1.2` in android/build.gradle
2. Updated com.google.firebase:firebase-... packages to versions as described in the notes
3. ./gradlew clean & build again

My code is this:
```
const fetchedUnits = await firebase.firestore().collection('boards').doc(this.currentBoard)
      .collection('posts')
      .orderBy('time', 'desc')
      .get()
```
I already tried:
- removing the `desc` parameter
- changing `'time'` to `firebase.firestore.FieldPath('time')`

### Environment

1. Application Target Platform:
Android SDK compileSdkVersion 26

2. Development Operating System:
macOs High Sierra 10.13.4

3. Build Tools:
buildToolsVersion "26.0.2"

4. React Native version:
0.54

5. RNFirebase Version:

4.2.0

6. Firebase Module:
Firestore

**Comment by okurichenko** [comment]
Same here.

### Environment
1. Application Target Platform: Android SDK compileSdkVersion 27

2. Development Operating System: macOs High Sierra 10.13.4

3. Build Tools: buildToolsVersion "27.0.3"

4. React Native version: 0.54.2

5. RNFirebase Version: 4.2.0

6. Firebase Module: Firestore

Tried to roll back to `4.0.7` version of RNFirebase, but it doesn't support Ref.onSnapshot listener, so it doesn't suit

**Comment by yuvalmarg** [comment]
Same here...
Did one of you guys find a workaround for this?

**Comment by yuvalmarg** [comment]
happens to me also using just .limit(x)

**Comment by OhadRon** [comment]
@yuvalmarg What I did is get the entire collection and sort it in my client. Obviously more expensive ($$$ + time + data) and only feasible if the collection is small. If you use `.limit()` you probably have no workaround at this point.

**Comment by yuvalmarg** [comment]
Yeah I'm doing something like that too at the moment.
I don't think the issue has to do with onSnapshot. It just happens when querying with limit or orderBy or both on a subcollection

**Comment by stigasoft** [comment]
+1
Did you find some other solutions except all data fetch and sort at the client?

**Comment by yuvalmarg** [comment]
@OhadRon @sourabhkheterpal I upgraded React-Native to 0.54.4 and it fixes the issue

**Comment by chrisbianca** [comment]
Yes, this a bug in React Native which was resolved in 0.54.4.
